core/src/main/scala/org/apache/spark/sql/hive/SnappyStoreHiveCatalog.scala

```diff
@@ -615,6 +622,44 @@ class SnappyStoreHiveCatalog(context: SnappyContext)
     }
   }
 
+  def alterTableToAddIndexProp(inTable: QualifiedTableName, indexName: String): Unit = {
+    alterTableLock.synchronized {
```

Comment:
Don't use indexName as Strings at this level. We should "normalize" the names to create QualifiedTableName's (can rename the class to just QualifiedName) at the very start. That takes care of case-sensitivity of names.